src/components/Table/components/Row.tsx

```diff
@@ -27,17 +26,17 @@ import { Css, Palette } from "src/Css";
 import { useComputed } from "src/hooks";
 import { shallowEqual } from "src/utils/shallowEqual";
 
-interface RowProps<R extends Kinded, S> {
+interface RowProps<R extends Kinded> {
   as: RenderAs;
-  columns: GridColumn<R>[];
+  columns: GridColumnWithId<R>[];
   row: GridDataRow<R>;
   style: GridStyle;
   rowStyles: RowStyles<R> | undefined;
   stickyHeader: boolean;
   stickyOffset: number;
   sortOn: SortOn;
-  sortState?: SortState<S>;
-  setSortKey?: (value: S) => void;
+  sortState?: SortState;
+  setSortKey?: (value: string) => void;
```

Comment:
Suggestion: Since the `RowImpl` is getting the `tableState` directly on line 70, we could probably remove these two props, and just access them from `tableState`?